packages/rest/src/parser.ts

```diff
@@ -38,11 +39,16 @@ export async function parseOperationArgs(
     operationSpec,
     request,
   );
+  const query = await requestBodyParser.loadRequestBodyIfNeeded(
```

Comment:
I doubt whether we can get the query using `requestBodyParser`, according to https://github.com/strongloop/loopback-next/blob/2e12925f2dcb89d2ef0262873615336446390386/packages/rest/src/body-parsers/body-parser.helpers.ts#L54, it eventually focuses and returns the `body` field in a request.

I also understand that there would be a good amount of logic that could be shared among body parsers and query parsers, I would suggest:
- We start from creating a working `query-parser.json.ts` like https://github.com/strongloop/loopback-next/blob/2e12925f2dcb89d2ef0262873615336446390386/packages/rest/src/body-parsers/body-parser.json.ts
- Refactor the common logic into utils so that body/query/header parser could share
- Implement the query parser and call its functions here.